py/selenium/webdriver/remote/remote_connection.py

```diff
@@ -487,87 +392,52 @@ def _request(self, method, url, body=None):
 
         parsed_url = parse.urlparse(url)
         headers = self.get_remote_connection_headers(parsed_url, self.keep_alive)
-
+        resp = None
         if self.keep_alive:
+
             if body and method != 'POST' and method != 'PUT':
                 body = None
-            try:
-                self._conn.request(method, parsed_url.path, body, headers)
-                resp = self._conn.getresponse()
-            except (httplib.HTTPException, socket.error):
-                self._conn.close()
-                raise
+            resp = self._conn.request(method, url, body=body, headers=headers)
 
             statuscode = resp.status
         else:
-            password_manager = None
-            if parsed_url.username:
-                netloc = parsed_url.hostname
-                if parsed_url.port:
-                    netloc += ":%s" % parsed_url.port
-                cleaned_url = parse.urlunparse((
-                    parsed_url.scheme,
-                    netloc,
-                    parsed_url.path,
-                    parsed_url.params,
-                    parsed_url.query,
-                    parsed_url.fragment))
-                password_manager = url_request.HTTPPasswordMgrWithDefaultRealm()
-                password_manager.add_password(None,
-                                              "%s://%s" % (parsed_url.scheme, netloc),
-                                              parsed_url.username,
-                                              parsed_url.password)
-                request = Request(cleaned_url, data=body.encode('utf-8'), method=method)
-            else:
-                request = Request(url, data=body.encode('utf-8'), method=method)
-
-            for key, val in headers.items():
-                request.add_header(key, val)
+            http = urllib3.PoolManager()
+            resp = http.request(method, url, body, headers=headers)
```

Comment:
This needs to have `body` set with keyword otherwise it's going into `fields`. Similar to what was done above in `keep_alive` version.

```
resp = http.request(method, url, body=body, headers=headers)
```

Reply:
I had the same issue both against Sauce Labs and when running against a docker-selenium instance.

@lmtierney's fix sorted that out.